cmd/router/main.go

```diff
@@ -133,69 +174,120 @@ func routeStep(nodeName string, graph v1alpha1.InferenceGraphSpec, input []byte,
 		}
 		// merge responses from parallel steps
 		response := map[string]interface{}{}
+		ensembleStepOutput := EnsembleStepOutput{}
 		for i, resultChan := range ensembleRes {
 			key := currentNode.Steps[i].StepName
 			if key == "" {
 				key = strconv.Itoa(i) // Use index if no step name
 			}
 			select {
-			case response[key] = <-resultChan:
+			case ensembleStepOutput = <-resultChan:
+				if ensembleStepOutput.StepStatusCode != 200 && currentNode.Steps[i].Dependency == v1alpha1.Hard {
```

Comment:
Here we can use the `isSuccessful` util to be consistent with other places

Reply:
Done